h/services/auth_ticket.py

```diff
@@ -1,42 +1,35 @@
-import base64
-import os
 from datetime import datetime, timedelta
 
 import sqlalchemy as sa
-from webob.cookies import SignedCookieProfile
 
-from h.models import AuthTicket
+from h.models import AuthTicket, User
 
 
-class AuthCookieService:
+class AuthTicketService:
     TICKET_TTL = timedelta(days=7)
 
     # We only want to update the `expires` column when the tickets `expires` is
     # at least one minute smaller than the potential new value. This prevents
     # that we update the `expires` column on every single request.
     TICKET_REFRESH_INTERVAL = timedelta(minutes=1)
 
-    def __init__(self, session, user_service, cookie):
+    def __init__(self, session, user_service):
         self._session = session
         self._user_service = user_service
-        self._cookie = cookie
         self._user = None
 
-    def verify_cookie(self):
+    def verify_ticket(self, userid: str | None, ticket_id: str | None) -> User | None:
         """
-        Get the authenticated user by cookie (if any).
+        Return the User object matching the given userid and ticket_id, or None.
 
-        :return: The logged in `User` or None
+        Verify that there is an unexpired AuthTicket in the DB matching the
+        given `userid` and `ticket_id` and if so return the corresponding User.
         """
 
         if self._user:
             # We've already vetted the user!
             return self._user
 
-        userid, ticket_id = self._get_cookie_value()
-        if not ticket_id:
```

Comment:
We saved a DB round trip with this before, I wonder why it didn't do if not user_id or not ticket_id

Reply:
Dunno. I don't think this often gets called with either `userid=None` or `ticket_id=None`. Sent a follow-up PR to add the optimisation anyway: https://github.com/hypothesis/h/pull/8876